src/manage.c

```diff
@@ -4162,6 +4163,29 @@ launch_osp_openvas_task (task_t task, target_t target, const char *scan_id,
         }
     }
 
+  /* Setup general task preferences */
+  max_checks = task_preference_value (task, "max_checks");
+  g_hash_table_insert (scanner_options,
+                       g_strdup ("max_checks"),
+                       max_checks ? max_checks : MAX_CHECKS_DEFAULT);
```

Comment:
MAX_CHECKS_DEFAULT should be g_strdup'ed because it will be g_freed when the hashtable is destroyed.